From 80f4743bc70ce2ebd516b8d5657ee5862f20d202 Mon Sep 17 00:00:00 2001
From: Nicholas Kazlauskas <nicholas.kazlauskas@amd.com>
Date: Fri, 11 Oct 2019 12:26:10 -0400
Subject: [PATCH 4224/4736] drm/amd/display: Free gamma after calculating
 legacy transfer function

[Why]
We're leaking memory by not freeing the gamma used to calculate the
transfer function for legacy gamma.

[How]
Release the gamma after we're done with it.

Signed-off-by: Nicholas Kazlauskas <nicholas.kazlauskas@amd.com>
Reviewed-by: Leo Li <sunpeng.li@amd.com>
---
 drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_color.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_color.c b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_color.c
index 5005eb07159e..2eb1313c0728 100644
--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_color.c
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_color.c
@@ -209,6 +209,7 @@ static int __set_legacy_tf(struct dc_transfer_func *func,
         res = mod_color_calculate_regamma_params(func, gamma, true, has_rom,
                                                  NULL);
 
+	dc_gamma_release(&gamma);
         return res ? 0 : -ENOMEM;
 }
 
-- 
2.17.1

